web/src/compose_validate.js

```diff
@@ -20,6 +20,10 @@ import * as settings_data from "./settings_data";
 import * as stream_data from "./stream_data";
 import * as util from "./util";
 
+// This variable tracks the action which invoked `compose_validate.validate`.
+// Default trigger is `send` and other possible trigger is 'schedule'.
+export let validation_trigger = "send";
```

Comment:
Global variable like this are a bad pattern; they require a lot of care to avoid ending up with them in the wrong state. We should just pass it as a parameter to `validate` and plumb it through the 3-4 functions from there to `show_wildcard_warnings`.

Reply:
Oh okay makes sense will do that.